examples/query/react/basic/src/tests/App.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { render, screen } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import App from '../App'
+import { setUpStore } from '../store'
+
+const server = setupServer(
+  rest.get('https://pokeapi.co/api/v2/pokemon/bulbasaur', (req, res, ctx) => {
+    const mockApiResponse = {
+      species: {
+        name: 'bulbasaur',
+      },
+      sprites: {
+        front_shiny:
+          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png',
+      },
+    }
+    return res(ctx.json(mockApiResponse))
+  })
+)
+
+describe('App', () => {
+  beforeAll(() => server.listen())
+
+  afterEach(() => server.resetHandlers())
+
+  afterAll(() => server.close())
```

Comment:
I think all that could go into a `setupMockServer` function:
```suggestion
// maybe in a separate file?
function setupMockServer() {
  const server = setupServer(
    rest.get('https://pokeapi.co/api/v2/pokemon/bulbasaur', (req, res, ctx) => {
      const mockApiResponse = {
        species: {
          name: 'bulbasaur',
        },
        sprites: {
          front_shiny:
            'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png',
        },
      }
      return res(ctx.json(mockApiResponse))
    })
  )
  beforeAll(() => server.listen())
  afterEach(() => server.resetHandlers())
  afterAll(() => server.close())
  return server;
}

  
const server = setupMockServer()
describe('App', () => {
```

Reply:
Agreed, this should be in a separate file.